### PR TITLE
Creating file management application using winforms

### LINKED ISSUE #5398: [Languages] Creating File Management Application using WinForms - C#
### Topic Suggestion
Creating File Management Application using WinForms -  C#  

#### Proposed article introduction

Naturally, people always need to record their data files on certain information they have every day and save them securely on a single file in the memory where they can retrieve it with much ease.
This is common to many, more so those people who record different data on operations they are running within their environment, for example, nurses who deal with patients records diseases they have treated on a particular day, medicines they have issued and the number of patients treated, etc. Considering a particular situation this person needs to have an efficient way to store the data and retrieve it.

In this tutorial, we shall demonstrate how you can create files and folders, write data to them in their interesting formats, read the data recorded and they can also delete the files initially recorded using WinForms. They can also access other data stored in their memory. We will demonstrate a step by step ways of creating the User Interface and their irrespective code for implementation on triggers.

#### Key takeaways 
1. Introduction to file management
2. Designing WinForms
3. Adding the necessary namespace
4. Setting up the file path
5. Implementing the create, write, read, delete operations
6. Debugging and publishing the application

#### Article quality
The proposed tutorial will be unique from other documentations on the internet since it allows users to record particular data. I will use simple language for every reader to understand the concepts using various illustrations. 

#### References
N/A

### PR DESCRIPTION
**Pull request by davismaina**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #5398

### CODE REVIEW

**Comment by srishilesh** [comment]
Hey @davismaina ,

The peer-reviewers are working towards completing existing reviews to take up more. Please wait patiently until a peer-reviewer pings you to get started.

Thanks.

**Comment by louisefindlay23** [comment]
Plagiarism check complete

File: content/articles/creating-file-management-application-using-winforms/index.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+This is crucial in this project solution since it will create a folder where the written data will reside in the memory in a precise manner. We can also specialize in the format that it will be written with; in this project, all our data will be stored in *`.txt files`*.
+
+Write the following code snippet.
+```C#
+ string path = Environment.CurrentDirectory + "/" + "Students.txt";// The path where files will be written and recorded as a txt file
+```
+#### Coding the controls
+> Note: To begin coding within a method, we need first double-click on the appropriate control for the filemanger.cs to load.
+
+In the **`<-`** (button previous) button, we add the following code that will be able to undo a file browsing in the web browser.
```
Comment (Jethro-magaji) [inline content/articles/creating-file-management-application-using-winforms/index.md:106]: This sentence and the one below needs some rephrasing.

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi! 
Well done for such an amazing article. However, there're a lot of corrections you need to make.
Please, run your article on tools like Grammarly and ProWritingAid for better editing.
Thanks.

**Pull request closed without merging**